src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CommandQueryExecutor.java

```diff
@@ -0,0 +1,106 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver.executor;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.jsonapi.service.cqldriver.CQLSessionCache;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Configured to execute queries for a specific command that relies on drive profiles MORE TODO
+ * WORDS
+ *
+ * <p>The following settings should be set via the driver profile:
+ *
+ * <ul>
+ *   <li><code>page-size</code>
+ *   <li><code>consistency</code>
+ *   <li><code>serial-consistency</code>
+ *   <li><code>default-idempotence</code>
+ * </ul>
+ */
+public class CommandQueryExecutor {
+
+  private static final String PROFILE_NAME_SEPARATOR = "-";
+
+  public enum QueryTarget {
+    TABLE,
+    COLLECTION;
+
+    final String profilePrefix;
+
+    QueryTarget() {
+      this.profilePrefix = name().toLowerCase();
```

Comment:
Is it safe to call accessor from constructor? I guess it has to be since nothing breaks.

Reply:
yes works :)